### PR TITLE
Integrate and test API for view campaign

### LINKED ISSUE #46: API Integration & View Campaign Card and Page Integration
**Description of feature/enhancement**
1. Integrate get campaign API to CampaignCard component and View Campaign page
2. Route view campaign page campaign(s) to respective campaigns page via dynamic routing

### PR DESCRIPTION
**Pull request by firwer**
Closes #46 .

**Overview of changes:**
1. Integrate & Test GET API for the CampaignCard component
2. Routing from View Campaign page to respective Campaign Card page via dynamic routing

Not Done:
1. Integrate & Test GET API for the view campaign component (Populate list of campaigns page) **_(Pending get all campaigns API fix)_**

Additional Note:
- Due to CORS policy, **API Fetch will not work in deployment,** I used Moesif Origin & CORS plugin to bypass the CORS error temporarily for development testing

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -8,6 +8,14 @@ const nextConfig = {
...
     ignoreBuildErrors: true,
   },
+  async rewrites() {
+    return [
+      {
+        source: '/api/:path*',
+        destination: 'http://localhost:8000/*',
+      },
+    ]
+  },
```
Comment (wilinetan) [inline next.config.js:18]: i didn't know we could this. is there a way to make this dynamic based on current environment?

**Review by wilinetan — COMMENTED** [review]
I created 2 custom hooks for getting all campaigns and 1 campaign. they are in the apis folder
I also created 2 new types:
- BackendCampaign: Currently what the backend campaign model is like (without country, donors etc)
- Campaign: what im expecting the Campaign type from backend will be (im not too sure about this but I just combined what the backend is currently sending and what we need for our FE like  donors, amount all those to make this)

think there's nothing more we can do for now because BE side not done yet.
maybe can work on the delete campaign API if its alr up?

**Comment by wilinetan** [comment]
@firwer I just had a call with xiaoen, and she say for the view campaign page can help to align the blue box to the Donations title and right side also add margins. can refer to the screenshot below, i think just need to add marginLeft and marginRight
Screen Shot 2023-02-22 at 9 22 17 PM

**Comment by firwer** [comment]
> @firwer I just had a call with xiaoen, and she say for the view campaign page can help to align the blue box to the Donations title and right side also add margins. can refer to the screenshot below, i think just need to add marginLeft and marginRight Screen Shot 2023-02-22 at 9 22 17 PM

Got it! Will make the adjustments

**Comment by firwer** [comment]
> @firwer I just had a call with xiaoen, and she say for the view campaign page can help to align the blue box to the Donations title and right side also add margins. can refer to the screenshot below, i think just need to add marginLeft and marginRight Screen Shot 2023-02-22 at 9 22 17 PM

Issue resolved, also I have commented out the useGetCampaigns() in the Campaign main page since the API is not ready, Axios is getting me 404 errors and invalid objects on react's end

**Review by wilinetan — APPROVED** [review]
The backend just updated the api so I have updated the code to get the data from them. LGTM thanks!

**Pull request merged**